ArduCopter/mode.h

```diff
@@ -739,6 +754,15 @@ class ModeGuided : public Mode {
 
 private:
 
+    enum GuidedMode : uint8_t {
+        Guided_TakeOff,
+        Guided_WP,
+        Guided_Velocity,
+        Guided_PosVel,
+        Guided_Angle,
+    };
+    GuidedMode mode() const { return guided_mode; }
```

Comment:
I think we should use a more specific name rather than, "mode".  How about get_submode()? or just submode()?